internal/lang/funcs/encoding.go

```diff
@@ -227,6 +228,33 @@ var URLEncodeFunc = function.New(&function.Spec{
 	},
 })
 
+// URLDecodeFunc constructs a function that applies URL decoding to a given string.
+var URLDecodeFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "str",
+			Type: cty.String,
+		},
+	},
+	Type:         function.StaticReturnType(cty.String),
+	RefineResult: refineNotNull,
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		rawQuery := args[0].AsString()
+
+		// Check for null bytes
+		if strings.Contains(rawQuery, "%00") {
+			return cty.UnknownVal(cty.String), fmt.Errorf("null bytes are not be allowed")
```

Comment:
tiny tiny nitpick:

```suggestion
			return cty.UnknownVal(cty.String), fmt.Errorf("null bytes (%00) are not supported in the input to urldecode")
```

Reply:
Is there any specific reason why null bytes should not be supported? Are we 100% certain nobody wants to decode binary data with this?

Reply:
I agree with Janos, Go doesn't use null-terminated strings, I don't think there's any reason to explicitly check for them.

Reply:
Please let me know the decision so I can delete it.

Reply:
We discussed it internally with @Yantrio and he agreed that we should support null bytes because of the above reasons.
Please remove it.